packages/internals/support/src/schema/component.schema.js

```diff
@@ -0,0 +1,27 @@
+const fileSchema = require('./file.schema');
+const fileCollectionSchema = require('./file-collection.schema');
+
+module.exports = {
+  $schema: 'http://json-schema.org/draft-06/schema#',
+  definitions: {
+    file: fileSchema,
+    fileCollection: fileCollectionSchema
+  },
+  type: 'object',
+  properties: {
+    src: {
+      $ref: '#/definitions/file'
```

Comment:
So I'm not sure referencing a File schema here is correct - it actually needs to be a `File` instance (as instantiating a component with a plain object that matches the file schema will still cause errors). I added custom `instanceof` checks to the validator on my pages branch last night so that stuff like this works:
```
src: {
   instanceof: 'File'
}
```

but I've not merged that in to V2 yet. Maybe let's leave as is for now and update when I've merged in the Validator updates?

Reply:
In this instance, the file schema will match either a File instance or file props, but a clear `instanceof` is certainly cleaner in general!

Reply:
Yeah the problem is that it _would_ match file props and so validate it as being ok - but it wouldn't actually work as the component requires an actual File instance